src/pages/menu-page/MenuPage.jsx

```diff
@@ -1,79 +1,136 @@
-import React, { useState } from 'react';
+import React, { Component } from 'react';
 import ProductCard from "../../components/product-card/ProductCard.jsx";
-import products from "../../assets/productInfo(temporary folder)/products.js";
 import PhoneNumberToolTip from "../../components/phone-number-tooltip/PhoneNumberToolTip.jsx";
 import './menuPage.css';
+import axios from 'axios';
 
+class MenuPage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isHovered: false,
+            activeButton: 'Desert',
+            meals: [],
+            loading: true,
+            error: null,
+            visibleCountCards: 6,
+        };
+    }
 
-function  MenuPage () {
-    const [isHovered, setIsHovered] = useState(false);
-    const [activeButton, setActiveButton] = useState('Desert');
-    const handleButtonClick = (buttonName) => {
-        if (activeButton !== buttonName) {
-            setActiveButton(buttonName)
+    componentDidMount() {
+        axios.get("https://65de35f3dccfcd562f5691bb.mockapi.io/api/v1/meals")
+            .then((response) => {
+                this.setState({meals: response.data, loading: false});
+            })
+
+            .catch((error) => {
+                this.setState({error: error.message, loading: false});
+            })
+    }
+
+    handleSeeMoreButton = () => {
+        this.setState((prevState) => ({
+            visibleCountCards: prevState.visibleCountCards + 6
+        }));
+    }
+
+    handleMouseEnter = () => {
+        this.setState({ isHovered: true });
+    };
+
+    handleMouseLeave = () => {
```

Comment:
That should be a part of Tooltip component, it has nothing to do with a MenuPage component